src/hotspot/share/ci/ciEnv.cpp

```diff
@@ -1208,15 +1207,19 @@ void ciEnv::register_method(ciMethod* target,
         }
       }
     }
-  }  // safepoints are allowed again
+  }
 
+  NoSafepointVerifier nsv;
```

Comment:
There is window after releasing all previous locks and here when nmethod could become dead. I don't see `post_compiled_method()` checking for that.
Also I don't see such `NoSafepointVerifier` in `JVMCIRuntime::register_method()`.

Reply:
The nmethod can only get unloaded at safepoint polls. There aren't any when unlocking. The reason the NSV isn't just crossing the entire scope, is because some locks like Compile_lock will block when we take the lock, just not when we unlock it.
Unfortunately I can't put a similar NSV in the JVMCI installer, because when you run JVMCI with native image, there is a setter that uses JNI, which will check for safepoints. However, that only happens if the nmethod failed to install, which doesn't really matter, but it does prevent the NSV from being in the same corresponding location there.